test/SILGen/moveonly.swift

```diff
@@ -1000,3 +1004,2094 @@ struct AddressOnlySetterTester : ~Copyable {
         set { fatalError() }
     }
 }
+
+/////////////////////
+// MARK: Subscript //
+/////////////////////
+
+// MARK: Getter Only
+
+public struct LoadableSubscriptGetOnlyTester : ~Copyable {
+    subscript(_ i: Int) -> AddressOnlyProtocol {
+        get {
+            fatalError()
+        }
+    }
+}
+
+// CHECK-LABEL: sil [ossa] @$s8moveonly047testSubscriptGetOnly_BaseLoadable_ResultAddressE4_VaryyF : $@convention(thin) () -> () {
+// CHECK: [[BOX:%.*]] = alloc_box $
+// CHECK: [[PROJECT:%.*]] = project_box [[BOX]]
+//
+// The get call
+// CHECK: [[ACCESS:%.*]] = begin_access [read] [unknown] [[PROJECT]]
+// CHECK: [[MARK:%.*]] = mark_must_check [no_consume_or_assign] [[ACCESS]]
+// CHECK: [[LOAD_BORROW:%.*]] = load_borrow [[MARK]]
+// CHECK: [[TEMP:%.*]] = alloc_stack $AddressOnlyProtocol
+// CHECK: [[TEMP_MARK:%.*]] = mark_must_check [consumable_and_assignable] [[TEMP]]
+// CHECK: apply {{%.*}}([[TEMP_MARK]], {{%.*}}, [[LOAD_BORROW]])
+// CHECK: apply {{%.*}}([[TEMP_MARK]])
+// CHECK: destroy_addr [[TEMP_MARK]]
+// CHECK: end_borrow [[LOAD_BORROW]]
+// CHECK: end_access [[ACCESS]]
+// CHECK: } // end sil function '$s8moveonly047testSubscriptGetOnly_BaseLoadable_ResultAddressE4_VaryyF'
+public func testSubscriptGetOnly_BaseLoadable_ResultAddressOnly_Var() {
+    var m = LoadableSubscriptGetOnlyTester()
+    m = LoadableSubscriptGetOnlyTester()
+    m[0].nonMutatingFunc()
```

Comment:
Since `m[0]` is implemented by a getter, the result is an independently consumable value. Should the formal access to `m` be shortened to end after `m[0]` returns? Does `let m0 = m[0]` work?

Reply:
It does. I added a test that validates that.